src/components/IGVAddTrackModal.js

```diff
@@ -0,0 +1,104 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
+import { form, h, label } from 'react-hyperscript-helpers'
+import { ButtonPrimary, IdContainer } from 'src/components/common'
+import { parseGsUri } from 'src/components/data/data-utils'
+import { TextInput, ValidatedInput } from 'src/components/input'
+import Modal from 'src/components/Modal'
+import * as Utils from 'src/libs/utils'
+
+
+const IGVAddTrackModal = ({ onDismiss, onSubmitTrack }) => {
+  const [value, setValue] = useState({
+    name: '',
+    url: '',
+    indexUrl: ''
+  })
+
+  const [urlInputTouched, setUrlInputTouched] = useState(false)
+
+  const urlError = Utils.cond(
+    [!value.url, () => 'A URL is required'],
+    [_.isEmpty(parseGsUri(value.url)), () => 'A gs:// URL is required'],
+    () => null
+  )
+
+  const isTrackValid = !urlError
```

Comment:
[Potentially a nit] Would

```suggestion
  const isTrackValid = !_.isNull(urlError)
```

be clearer/safer? Probably not a big deal in this case but when I see JS coercion of a non-boolean value into boolean with assumptions (e.g. it's `null` when there's an error, in this case), I wonder if that value could be modified to be something else (e.g. `0`) down the road, falsefying the assumption without noticing its implications.

Reply:
Changed in https://github.com/DataBiosphere/terra-ui/pull/3282/commits/c2b065d1e873c55682e8d821768028e960a6e4f3.